src/base/bittorrent/session.cpp

```diff
@@ -2408,6 +2411,24 @@ void Session::saveResumeData()
     }
 }
 
+void Session::saveTorrentsQueue()
+{
+    QMap<int, QString> queue; // Use QMap since it should be ordered by key
+    for (const TorrentHandle *torrent : torrents()) {
+        // We require actual (non-cached) queue position here!
+        const int queuePos = torrent->nativeHandle().queue_position();
+        if (queuePos >= 0)
+            queue[queuePos] = torrent->hash();
+    }
+
+    QByteArray data;
+    QTextStream out {&data};
```

Comment:
Personally I would omit using it, although it is convenience, it is quite heavy IIRC, besides you only have 1 one using it.

Reply:
>Why should I do it in a less convenient way (even if I need it in one place? 

Are you sure there won't be any additional/unwanted performance penalty for using it?

Reply:
`data += (hash + '\n');` isn't that much unreadable.

Reply:
hash is of type **InfoHash** so it will be the following:
```c++
data += (static_cast<QString>(hash).toLatin1() + '\n');
```
Changed anyway.